source/engine/graphics/components/SkeletalAnimator.cpp

```diff
@@ -1,45 +1,204 @@
 #include "graphics/SkeletalAnimator.h"
-#include "graphics/MeshRenderer.h"
-#include "graphics/ToonRenderer.h"
-#include "ecs/Registry.h"
 
 namespace nc::graphics
 {
-SkeletalAnimator::SkeletalAnimator(Entity entity, std::string meshUid, std::string animationUid)
+SkeletalAnimator::SkeletalAnimator(Entity entity, const std::string& meshUid, const std::string& animationUid)
     : ComponentBase{entity},
-      m_meshUid{std::move(meshUid)},
-      m_onPlayStateChanged{},
-      m_initialState{anim::State{anim::Initial{std::move(animationUid)}}},
-      m_initialStateInitialized{}{}
+      m_states{MaxStates},
+      m_activeState{m_states.Insert(anim::State{anim::Root{animationUid}})},
+      m_immediateState{},
+      m_immediateStateSet{},
+      m_rootStateSet{},
+      m_meshUid{meshUid},
+      m_onStateChanged{}
+{
+}
 
 void SkeletalAnimator::Execute()
 {
-    if (ExecuteInitialState())
+    if (ExecuteRootState()) { return; }
+    if (ExecuteImmediateState()) { return; }
+    if (ExecuteExitState()) { return; }
+
+    for (auto childStateHandle : m_states.Get(m_activeState)->successors)
```

Comment:
`any_of`?

Reply:
It actually makes the code more verbose and less understandable, in this case.